llvm/test/CodeGen/RISCV/rvv/sink-splat-operands.ll

```diff
@@ -1176,9 +1176,9 @@ define void @sink_splat_ashr_scalable(ptr nocapture %a) {
 ; CHECK-NEXT:    andi a3, a1, 1024
 ; CHECK-NEXT:    xori a1, a3, 1024
 ; CHECK-NEXT:    slli a4, a4, 1
-; CHECK-NEXT:    vsetvli a5, zero, e32, m2, ta, ma
 ; CHECK-NEXT:    mv a5, a0
```

Comment:
This one is an interesting example.  The destination vreg is presumably dead, and by placing it later in the block we have higher register pressure.  My guess is that these moves come from phi elimination?  If so, maybe we have a slightly different move backwards to reduce register pressure lurking here?

Reply:
To be explicit, I'm fine with this landing without the backwards heuristic as you've removed the diffs by taking them eagerly in another patch, but I would like to see us follow up on this after this series has landed.  I think the scheduling to reduce register pressure is interesting here, and if the scheduler doesn't manage that on it's own after the whole patch sequence, we should look into this more.

Reply:
I gave a very quick stab at moving the vsetvli up before the previous GPR def (so the dead def is immediately reused), and we're able to reuse the GPR and remove the spill in this case. https://github.com/lukel97/llvm-project/commit/fe81aa40df42a53bda6b407888cb2322618f23f5

But on SPEC 2017 it actually seems to increase the number of spills by 0.2%, both on main and when applied to this PR. So it doesn't look like its actually removing any spills that may have been introduced by this PR. 

My guess is that since we have to update LiveIntervals that it's perturbing something in regalloc ever so slightly, or it might actually be moving the vsetvli up into a higher pressure region and backfiring.

This increase in register pressure seems small though (it's limited to just the single instruction slot and it's in the scalar domain, so hopefully less likely to spill than a vector), are we considering it something that needs fixed in this PR?

Reply:
Woops, your reply came in as I was typing my comment so I didn't see it, I agree we should follow up on this. I did try inducing a spill in a local test case and I was surprised to see that the scheduler doesn't actually move the vsetvli to avoid the spill. And I presume this is not specific to vsetvlis but for any dead def.